master/pyext/src/validation/sas.py

```diff
@@ -66,10 +68,20 @@ def get_data_from_SASBDB(self) -> dict:
         for code in self.get_SASBDB_code():
             if 'None' not in str(code):
                 url_f = self.sasentry+code+'.json'
-                response = requests.get(url_f, data={'key': 'value'})
-                if response.status_code != 200:
-                    print(
-                        "Error....unable to fetch data from SASBDB, please check the entry ID")
+
+                # Check if we already requested the data
+                try:
+                    response = self.SASBDB_cache[url_f]
+                except KeyError:
+                    response = requests.get(url_f, data={'key': 'value'})
```

Comment:
Consider using `functools.lru_cache` here, or writing a function/class. There's a lot of duplicated code here.